### DISCUSSION TITLE
New data model not detected

### DISCUSSION THREAD

**Discussion by vinitpradhn18** [apache/superset#17656] [Q&A / Help]
I have added a new model. However, whenever I start docker, superset_init container does not detect new models and create the respective table in the Postgres database. I don't  see alembic revision file created as well. Do I need to manually create a revision file? If so how do I do that. I am not sure if I am missing anything.

Any help would be appreciated. Thanks

**Comment by nytai** [comment]
You need to generate an alembic migration file here https://github.com/apache/superset/tree/master/superset/migrations/versions

check alembic docs for how to

**Reply by nikhil-kuyya-talentas** [reply]
does this cause issue with upgrade with superset.

**Reply by nikhil-kuyya-talentas** [reply]
not able to run alembic in docker-container(superset_app) virtualenvironment shows error
bad interpreter: `No such file or directory`
any taughts on using alembic in container.

i have tried in locally (without docker) shows error
`RuntimeError: Working outside of application context.`

 Thank you in advance

**Reply by nikhil-kuyya-talentas** [reply]
@vinitpradhn18 if you were able to create the revision files, could you help me steps if possible because i am facing some issues like mentioned above.i have experience with frontend but not much with python

**Reply by vinitpradhn18** [reply]
exec into `superset_app` container and then you can run `superset db migrate -m <commit message>`. This will create the revision file if schema changes are detected. You can verify the file in `/superset/migrations/versions` folder
Then run `superset db upgrade` or start `superset_init` container
